File: 7-kyu/how-long-should-you-cook-this-for-/javascript/solution.js
const cookingTime = (neededPower, minutes, seconds, power) => {
  const actualNeededPower = Number(neededPower.slice(0, -1));
  const actualPower = Number(power.slice(0, -1));
  const totalSeconds = minutes * 60 + seconds;
  const neededSeconds = Math.ceil(totalSeconds * (actualNeededPower / actualPower));
  const actualMinutes = Math.floor(neededSeconds / 60);
  const actualSeconds = neededSeconds % 60;
  return `${actualMinutes} minutes ${actualSeconds} seconds`;
};

File: 7-kyu/how-long-should-you-cook-this-for-/python/solution.py
from math import ceil, floor

def cooking_time(needed_power, minutes, seconds, power):
    pwr = int(power[:-1])
    needed_pwr = int(needed_power[:-1])
    ratio = needed_pwr / pwr
    total_secs = ceil((minutes * 60 + seconds) * ratio)
    return f'{floor(total_secs / 60)} minutes {total_secs % 60} seconds'
